### PR TITLE
GCI: Add two GCI specific metadata pairs

### PR DESCRIPTION
**Pull request by andyzheng0831**
This PR adds two GCI specific metadata pairs when using GCI image.

(1) "gci-update-strategy": by default the GCI in-place updater is enabled. It means that when a new image is released, the instance on the old image will be upgraded to the new image. In this change, we turn it off;

(2) "gci-ensure-gke-docker": GCI is built with two versions of docker. When this metadata is set to "true", the version satisfying kubernetes qualification will be used. Setting this metadata prevents from using incorrect docker version.

### CODE REVIEW

**Comment by andyzheng0831** [comment]
@roberthbailey @dchen1107 @zmerlynn please review it. We also need it in 1.2.X releases. Please set needed labels and milestone.

cc/ @fabioy @wonderfly FYI

**Comment by andyzheng0831** [comment]
@k8s-bot unit test this issue #24704

File: cluster/gce/trusty/helper.sh [hunk 1]
```diff
@@ -18,17 +18,45 @@
 
 # The configuration is based on upstart, which is in Ubuntu up to 14.04 LTS (Trusty).
 # Ubuntu 15.04 and above replaced upstart with systemd as the init system.
-# Consequently, the configuration cannot work on these images.
+# Consequently, the configuration cannot work on these images. In release-1.2 branch,
+# GCI and Trusty share the configuration code. We have to keep the GCI specific code
```
Comment (roberthbailey) [inline cluster/gce/trusty/helper.sh]: Why do we have to worry about keeping this code separate on the master branch? Is it just to make cherry picks easier?

Comment (andyzheng0831) [inline cluster/gce/trusty/helper.sh]: I am preparing for a cluster/gce/gci dir which serves gci only. But such a big change will be in 1.3 only. For 1.2.X gci will still use the code in cluster/gce/trusty. Later if a PR breaking GCI is cherry picked to release-1.2, we will have to make 3 PR: one for fixing master branch cluster/gce/gci, one for fixing master branch cluster/gce/trusty, and another for cherry-pick into cluster/gce/trusty in release-1.2. If we only keep gci-specific code in master branch, it is hard to cherry-pick into release-1.2 as there is no dir cluster/gce/gci

**Comment by dchen1107** [comment]
LGTM

**Comment by dchen1107** [comment]
From the code, I don't think your pr breaks the trusty support, but want to double check it.

File: cluster/gce/trusty/helper.sh (hunk 1 above)
Comment (roberthbailey) [inline cluster/gce/trusty/helper.sh]: Ok, so it's to make cherry picks easier. That's what I thought.

**Comment by andyzheng0831** [comment]
No, it does not break trusty. I had run e2e on 4 scenarios: (1) Master + nodes on trusty; (2) Master on trusty, nodes on ContainerVM; (3) Master + nodes on GCI; (4) Master on GCI, nodes on ContainerVM. They are all green.

**Comment by andyzheng0831** [comment]
@k8s-bot unit test this issue #23433

**Pull request merged**